sdk/python/tests/conftest.py

```diff
@@ -164,129 +161,119 @@ def start_test_local_server(repo_path: str, port: int):
-            )
-
-            cls.container.start()
-            log_string_to_wait_for = "SERVER STARTED"
-            wait_for_logs(
-                container=cls.container, predicate=log_string_to_wait_for, timeout=30
-            )
-            cls.is_running = True
-        return cls.container
-
-    @classmethod
-    def teardown(cls):
-        if cls.container:
-            cls.container.stop()
+# @pytest.fixture(
+#     params=FULL_REPO_CONFIGS, scope="session", ids=[str(c) for c in FULL_REPO_CONFIGS]
+# )
+# def environment(request, worker_id: str, trino_fixture):
+#     if "TrinoSourceCreator" in request.param.offline_store_creator.__name__:
+#         e = construct_test_environment(
+#             request.param,
+#             worker_id=worker_id,
+#             offline_container=trino_fixture.get_singleton(),
+#         )
+#     else:
+#         e = construct_test_environment(request.param, worker_id=worker_id)
+#     proc = Process(
+#         target=start_test_local_server,
+#         args=(e.feature_store.repo_path, e.get_local_server_port()),
+#         daemon=True,
+#     )
+#     if e.python_feature_server and e.test_repo_config.provider == "local":
+#         proc.start()
+#         # Wait for server to start
+#         time.sleep(3)
+#
+#     def cleanup():
+#         e.feature_store.teardown()
+#         if proc.is_alive():
+#             proc.kill()
+#         if e.online_store_creator:
+#             e.online_store_creator.teardown()
+#
+#     request.addfinalizer(cleanup)
+#
+#     return e
 
 
 @pytest.fixture(scope="session")
-def trino_fixture(request):
-    def teardown():
-        TrinoContainerSingleton.teardown()
-
-    request.addfinalizer(teardown)
-    return TrinoContainerSingleton
-
-
-@pytest.fixture(
-    params=FULL_REPO_CONFIGS, scope="session", ids=[str(c) for c in FULL_REPO_CONFIGS]
-)
-def environment(request, worker_id: str, trino_fixture):
-    if "TrinoSourceCreator" in request.param.offline_store_creator.__name__:
-        e = construct_test_environment(
-            request.param,
-            worker_id=worker_id,
-            offline_container=trino_fixture.get_singleton(),
+def environment(request, worker_id):
+    e = construct_test_environment(request.param, worker_id=worker_id)
+
+    yield e
+
+    e.feature_store.teardown()
+    e.data_source_creator.teardown()
+    if e.online_store_creator:
+        e.online_store_creator.teardown()
+
+
+_config_cache = {}
+
+
+def pytest_generate_tests(metafunc):
+    if "environment" in metafunc.fixturenames:
+        markers = {m.name: m for m in metafunc.definition.own_markers}
+
+        if "universal_offline_stores" in markers:
+            offline_stores = AVAILABLE_OFFLINE_STORES
+        else:
+            # default offline store for testing online store dimension
+            offline_stores = [("local", FileDataSourceCreator)]
+
+        online_stores = None
+        if "universal_online_stores" in markers:
```

Comment:
So I need a name here that would represent that specific tests works with all offline stores (`universal_offline_stores` mark currently) or all online stores (`universal_online_stores` mark). Any ideas for a better name?

Reply:
I meant the `pytest.mark.universal` flag (since it seems like this the `universal_online_stores` and `universal_offline_stores` together would represent the same thing that current flag is supposed to represent).